pkg/strategy/autobuy/strategy.go

```diff
@@ -0,0 +1,146 @@
+package autobuy
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/c9s/bbgo/pkg/bbgo"
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	indicatorv2 "github.com/c9s/bbgo/pkg/indicator/v2"
+	"github.com/c9s/bbgo/pkg/strategy/common"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/robfig/cron/v3"
+	"github.com/sirupsen/logrus"
+)
+
+const ID = "autobuy"
+
+var log = logrus.WithField("strategy", ID)
+
+func init() {
+	bbgo.RegisterStrategy(ID, &Strategy{})
+}
+
+type Strategy struct {
+	*common.Strategy
+
+	Environment *bbgo.Environment
+	Market      types.Market
+
+	Symbol           string                  `json:"symbol"`
+	CronExpression   string                  `json:"cronExpression"`
+	Threshold        fixedpoint.Value        `json:"threshold"`
+	BollingerSetting *types.BollingerSetting `json:"bollingerSetting"`
+	DryRun           bool                    `json:"dryRun"`
+
+	bbgo.QuantityOrAmount
+
+	boll *indicatorv2.BOLLStream
+	cron *cron.Cron
+}
+
+func (s *Strategy) Initialize() error {
+	if s.Strategy == nil {
+		s.Strategy = &common.Strategy{}
+	}
+	return nil
+}
+
+func (s *Strategy) ID() string {
+	return ID
+}
+
+func (s *Strategy) InstanceID() string {
+	return fmt.Sprintf("%s:%s", ID, s.Symbol)
+}
+
+func (s *Strategy) Validate() error {
+	if err := s.QuantityOrAmount.Validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Strategy) Subscribe(session *bbgo.ExchangeSession) {
+	session.Subscribe(types.KLineChannel, s.Symbol, types.SubscribeOptions{Interval: s.BollingerSetting.Interval})
+}
+
+func (s *Strategy) Run(ctx context.Context, _ bbgo.OrderExecutor, session *bbgo.ExchangeSession) error {
+	s.Strategy.Initialize(ctx, s.Environment, session, s.Market, ID, s.InstanceID())
+
+	s.boll = session.Indicators(s.Symbol).BOLL(s.BollingerSetting.IntervalWindow, s.BollingerSetting.BandWidth)
+
+	session.UserDataStream.OnBalanceUpdate(func(balances types.BalanceMap) {
```

Comment:
BalanceUpdate could be triggered very fast, you will need to use rate limiter

Reply:
I changed to do this on order filled.